2/answers.txt

```diff
@@ -0,0 +1,231 @@
+
+A git repository is a essentially a version control system that allows multiple developers to work simultaneously on source code and effectively allows for updates, merging, compare and other similar functions like cloning and forking to provide
+developers with the tools they need to work collaboratively. 
+
+Cloning simply copies the repository from its remote source to a location locally. Becoming standalone in the process and disconnected from the source content.
+
+The remote in this instance is the location that is a copy of the original branch, whereas the origin is the centralsied point from which the most up to date version of a project would be distributed from.
+
+Q5: What is the benefit of forking a repository and submitting pull requests vs sharing repository access and pushing directly to master?
+
+I believe this is because its far safer for you to have your own copy, with which you can experiment freely with, and risk nothing other than your copy, then when you are satisfied that your work does not break anything or fail.
+You can submit a pull request for the original author to view the changes before integrating them with the original source code. Protecting the project from potential errors.
+
+Sharing repository access and pushing directly to the master could result in unforseen changes, potential for multiple changes from different users to conflict, and other such erroneous behaviour.
+
+Q6: Why are git branches useful? how would you organise a repository to best use branches?
+
+Branches are useful because it allows you to move the head and thus pointer to different instances of the work you are doing, enabling you to seperate out paths of changes, whether thats fixing mulitple seperate issues. Or coming up
+with multiple solutions for the same problem, branching these allows you to pick and choose between them and merge the ones that are useful, and merge changes with the master branch.
+
+To organise them I would simply create a branch for each change you wish to make, roughly speaking, so if you were to make a change to the code to solve a particular bug or functionality requirement then you would title that branch accordingly.
+
+Allowing you to compare and merge those changes individually and check for issues that may arise or errors that are made before combining them into the master branch.
+
+Q7: Why is it important to frequently commit and push your changes to a remote repository (eg your forked repo on GitHub)?
+
+Frequent commit and push will safeguard your work against loss of data if the original is somehow compromised, as well as ensuring that with smaller consistent changes, anything that causes
+a problem elsewhere or breaks something is easily identifiable, committing a lot of work at once increases the chance of a conflict with code that is submitted and will make it harder to
+troubleshoot an error should one arise.
```

Comment:
Frequent commit and push is more achievable when working in a branch, as when the branch is merged into master the hundreds of individual commits can be squashed into a single commit. The branch is then usually deleted.
This results in a clean git history.

In a public repo, each commit should have a meaningful purpose and a meaningful description.

In a forked/private repo, you can commit more frequently.
The project owner will probably squash your commits when they merge your pull request. They will likely be observing your code changes as a whole rather than the individual commits.